pkg/neg/syncers/utils.go

```
@@ -382,17 +449,17 @@ func validatePod(pod *apiv1.Pod, nodeLister cache.Indexer) bool {
 	// Terminal Pod means a pod is in PodFailed or PodSucceeded phase
 	phase := pod.Status.Phase
 	if phase == apiv1.PodFailed || phase == apiv1.PodSucceeded {
-		klog.V(2).Info("Pod %s/%s is a terminal pod with status %v, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, phase)
+		klog.V(2).Infof("Pod %s/%s is a terminal pod with status %v, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, phase)
 		return false
 	}
 	obj, exists, err := nodeLister.GetByKey(pod.Spec.NodeName)
 	if err != nil || !exists {
-		klog.V(2).Info("Pod %s/%s corresponds to a non-existing node %s, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, pod.Spec.NodeName)
+		klog.V(2).Infof("Pod %s/%s corresponds to a non-existing node %s, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, pod.Spec.NodeName)
 		return false
 	}
 	_, isNode := obj.(*apiv1.Node)
 	if !isNode {
-		klog.V(2).Info("Pod %s/%s does not correspond to a valid node resource, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
+		klog.V(2).Infof("Pod %s/%s does not correspond to a valid node resource, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
 		return false
 	}
```

perhaps all of these should return an error instead. And the caller will then log the error. Its a little weird that the helper function is logging when the outer function will be "skipping"

---

Updated. Now we log at outer functions based on the error.